File: viz/ggplot/theme_clinical.R
# A custom ggplot2 theme with a clinical and minimalist style.
#
# To use, save this file and then source it in your R script:
# source("path/to/theme_clinical.R")

library(ggplot2)
library(ggthemes)

# 1. COLOR PALETTE
# -----------------------------------------------------------------------------

clinical_colors <- c(
  "#3c5488",
  "#e64b35",
  "#00a087",
  "#4dbbd5",
  "#e18727"
)

# 2. CUSTOM THEME
# -----------------------------------------------------------------------------

theme_clinical <- function(base_size = 13, base_family = "sans") {
  # Start with theme_minimal() as a base and add modifications
  (theme_foundation(base_size = base_size, base_family = base_family)
  + theme(
      # General plot appearance
      plot.background = element_rect(fill = "white", colour = NA),
      plot.title = element_text(
        face = "bold",
        size = rel(1.44), # 20% larger than axis titles
        hjust = 0, # Left-align title
        margin = margin(b = 10)
      ),
      plot.subtitle = element_text(
        size = rel(1.2), # Same size as axis titles
        hjust = 0, # Left-align subtitle
        margin = margin(b = 10)
      ),
      plot.caption = element_text(
        size = rel(0.8),
        hjust = 1, # Right-align caption
        margin = margin(t = 10)
      ),

      # Panel appearance
      panel.background = element_rect(fill = "#EEEEEE", colour = "#000000"),
      panel.grid.major = element_line(colour = "grey60", linetype = "solid", linewidth = 0.5),
      panel.grid.minor = element_line(colour = "grey80", linetype = "solid", linewidth = 0.5),

      # Axis appearance
      axis.ticks = element_line(colour = "black", linetype = "solid", linewidth = 0.5), # Thin axis ticks
      axis.text = element_text(size = rel(1), colour = "black"), # Base size
      axis.title = element_text(size = rel(1.2), colour = "black"), # 20% larger than axis text

      # Legend appearance
      legend.position = "right",
      legend.background = element_rect(fill = "white", colour = NA), # Light grey background
      legend.key = element_rect(fill = "white", colour = NA), # Match panel background
      legend.title = element_text(face = "bold"),

      # Facet appearance
      strip.background = element_rect(fill = "grey90", colour = NA),
      strip.text = element_text(
        face = "bold",
        size = rel(1.1),
        margin = margin(t = 5, b = 5)
      )
    ))
}

File: viz/python/plotnine/theme_clinical.py
import plotnine as p9

clinical_colors = ["#F4D58D", "#8DB4D2", "#958ED2", "#D2958E", "#8ED2C2"]

def theme_clinical(base_size=13, base_family='sans'):
    """
    A custom plotnine theme with a clinical and minimalist style.
    This theme also includes default color and fill scales using the clinical_colors palette.

    To use:
    from theme_clinical import theme_clinical
    plot + theme_clinical()
    """
    theme = p9.themes.theme_minimal(base_size=base_size, base_family=base_family) + p9.themes.theme(
        # General plot appearance
        plot_background=p9.element_rect(fill="white", colour=None),
        plot_title=p9.element_text(
            weight="bold",
            size=base_size * 1.44,
            ha="left",
            margin={'b': 10}
        ),
        plot_subtitle=p9.element_text(
            size=base_size * 1.2,
            ha="left",
            margin={'b': 10}
        ),
        plot_caption=p9.element_text(
            size=base_size * 0.8,
            ha="right",
            margin={'t': 10}
        ),

        # Panel appearance
        panel_background=p9.element_rect(fill="#EEEEEE", colour=None),
        panel_border=p9.element_rect(colour='black', fill=None, size=0.5),
        panel_grid_major=p9.element_line(colour="#999999", linetype="solid", size=0.5),
        panel_grid_minor=p9.element_line(colour="#CCCCCC", linetype="solid", size=0.5),

        # Axis appearance
        axis_ticks=p9.element_line(colour="black", linetype="solid", size=0.5),
        axis_text=p9.element_text(size=base_size, colour="black"),
        axis_title=p9.element_text(size=base_size * 1.2, weight="bold", colour="black"),

        # Legend appearance
        legend_position="right",
        legend_background=p9.element_blank(),
        legend_key=p9.element_blank(),
        legend_title=p9.element_text(weight="bold"),

        # Facet appearance
        strip_background=p9.element_rect(fill="#E5E5E5", colour=None),
        strip_text=p9.element_text(
            weight="bold",
            size=base_size * 1.1,
            margin={'t': 5, 'b': 5}
        )
    )

    scales = [
        p9.scale_color_manual(values=clinical_colors),
        p9.scale_fill_manual(values=clinical_colors)
    ]

    return [theme] + scales
